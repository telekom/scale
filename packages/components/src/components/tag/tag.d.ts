export declare class Tag {
    /** (optional) Tag variant */
    variant?: string;
    /** (optional) Tag pill */
    pill?: boolean;
    /** (optional) Tag on an <a> element */
    link?: string;
    private getCssClassMap;
    render(): any;
}
