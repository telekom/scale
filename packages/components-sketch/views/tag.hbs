{{!--
Scale https://github.com/telekom/scale

Copyright (c) 2021 Egor Kirpichev and contributors, Deutsche Telekom AG

This Source Code Form is subject to the terms of the Mozilla Public
License, v. 2.0. If a copy of the MPL was not distributed with this
file, You can obtain one at https://mozilla.org/MPL/2.0/.
--}}

<div data-sketch-artboard="Chips/Tags" class="artboard">
  <h1 class="title--artboard">Tags</h1>

  <h2 class="title--artboard-section">
    Interactive (Clear tag) – Black – Large
  </h2>
  <div class="agent-states--grid" data-category="Interactive">
    <div class="agent-states--item">
      <p class="agent-states--label">:enabled</p>
      <scale-tag data-sketch-key="tag-0" dismissable>Tag Label</scale-tag>
    </div>
    <div class="agent-states--item">
      <p class="agent-states--label">:hover</p>
      <scale-tag data-sketch-key="tag-1" dismissable data-sketch-state="hover"
        >Tag Label</scale-tag
      >
    </div>
    <div class="agent-states--item">
      <p class="agent-states--label">:focus</p>
      <scale-tag data-sketch-key="tag-2" dismissable data-sketch-state="focus"
        >Tag Label</scale-tag
      >
    </div>
    <div class="agent-states--item">
      <p class="agent-states--label">:active/:selected</p>
      <scale-tag data-sketch-key="tag-3" dismissable data-sketch-state="active"
        >Tag Label</scale-tag
      >
    </div>
    <div class="agent-states--item">
      <p class="agent-states--label">:disabled</p>
      –
    </div>
  </div>

  <h2 class="title--artboard-section">
    Interactive (Clear tag) – Black – Small
  </h2>
  <div class="agent-states--grid" data-category="Interactive">
    <div class="agent-states--item">
      <p class="agent-states--label">:enabled</p>
      <scale-tag data-sketch-key="tag-5" size="small" dismissable
        >Tag Label</scale-tag
      >
    </div>
    <div class="agent-states--item">
      <p class="agent-states--label">:hover</p>
      <scale-tag
        data-sketch-key="tag-6"
        size="small"
        dismissable
        data-sketch-state="hover"
        >Tag Label</scale-tag
      >
    </div>
    <div class="agent-states--item">
      <p class="agent-states--label">:focus</p>
      <scale-tag
        data-sketch-key="tag-7"
        size="small"
        dismissable
        data-sketch-state="focus"
        >Tag Label</scale-tag
      >
    </div>
    <div class="agent-states--item">
      <p class="agent-states--label">:active/:selected</p>
      <scale-tag
        data-sketch-key="tag-8"
        size="small"
        dismissable
        data-sketch-state="active"
        >Tag Label</scale-tag
      >
    </div>
    <div class="agent-states--item">
      <p class="agent-states--label">:disabled</p>
      –
    </div>
  </div>

  <h2 class="title--artboard-section">
    Interactive (Clear tag) – Colored – Large
  </h2>
  <div class="agent-states--grid" data-category="Interactive">
    <div class="agent-states--item">
      <p class="agent-states--label">:enabled</p>
      <scale-tag data-sketch-key="tag-10" variant="secondary" dismissable
        >Tag Label</scale-tag
      >
    </div>
    <div class="agent-states--item">
      <p class="agent-states--label">:hover</p>
      <scale-tag
        data-sketch-key="tag-11"
        variant="secondary"
        dismissable
        data-sketch-state="hover"
        >Tag Label</scale-tag
      >
    </div>
    <div class="agent-states--item">
      <p class="agent-states--label">:focus</p>
      <scale-tag
        data-sketch-key="tag-12"
        variant="secondary"
        dismissable
        data-sketch-state="focus"
        >Tag Label</scale-tag
      >
    </div>
    <div class="agent-states--item">
      <p class="agent-states--label">:active/:selected</p>
      <scale-tag
        data-sketch-key="tag-13"
        variant="secondary"
        dismissable
        data-sketch-state="active"
        >Tag Label</scale-tag
      >
    </div>
    <div class="agent-states--item">
      <p class="agent-states--label">:disabled</p>
      –
    </div>
  </div>

  <h2 class="title--artboard-section">
    Interactive (Clear tag) – Colored – Small
  </h2>
  <div class="agent-states--grid" data-category="Interactive">
    <div class="agent-states--item">
      <p class="agent-states--label">:enabled</p>
      <scale-tag
        data-sketch-key="tag-15"
        variant="secondary"
        size="small"
        dismissable
        >Tag Label</scale-tag
      >
    </div>
    <div class="agent-states--item">
      <p class="agent-states--label">:hover</p>
      <scale-tag
        data-sketch-key="tag-16"
        variant="secondary"
        size="small"
        dismissable
        data-sketch-state="hover"
        >Tag Label</scale-tag
      >
    </div>
    <div class="agent-states--item">
      <p class="agent-states--label">:focus</p>
      <scale-tag
        data-sketch-key="tag-17"
        variant="secondary"
        size="small"
        dismissable
        data-sketch-state="focus"
        >Tag Label</scale-tag
      >
    </div>
    <div class="agent-states--item">
      <p class="agent-states--label">:active/:selected</p>
      <scale-tag
        data-sketch-key="tag-18"
        variant="secondary"
        size="small"
        dismissable
        data-sketch-state="active"
        >Tag Label</scale-tag
      >
    </div>
    <div class="agent-states--item">
      <p class="agent-states--label">:disabled</p>
      –
    </div>
  </div>

  <h2 class="title--artboard-section">Not Interactive – Black – Large</h2>
  <div class="agent-states--grid" data-category="Static">
    <div class="agent-states--item">
      <p class="agent-states--label">:enabled</p>
      <scale-tag data-sketch-key="tag-20">Tag Label</scale-tag>
    </div>
    <div class="agent-states--item">
      <p class="agent-states--label">:hover</p>
      –
    </div>
    <div class="agent-states--item">
      <p class="agent-states--label">:focus</p>
      –
    </div>
    <div class="agent-states--item">
      <p class="agent-states--label">:active/:selected</p>
      –
    </div>
    <div class="agent-states--item">
      <p class="agent-states--label">:disabled</p>
      –
    </div>
  </div>

  <h2 class="title--artboard-section">Not Interactive – Black – Small</h2>
  <div class="agent-states--grid" data-category="Static">
    <div class="agent-states--item">
      <p class="agent-states--label">:enabled</p>
      <scale-tag data-sketch-key="tag-21" size="small">Tag Label</scale-tag>
    </div>
    <div class="agent-states--item">
      <p class="agent-states--label">:hover</p>
      –
    </div>
    <div class="agent-states--item">
      <p class="agent-states--label">:focus</p>
      –
    </div>
    <div class="agent-states--item">
      <p class="agent-states--label">:active/:selected</p>
      –
    </div>
    <div class="agent-states--item">
      <p class="agent-states--label">:disabled</p>
      –
    </div>
  </div>

  <h2 class="title--artboard-section">Not Interactive – Colored – Large</h2>
  <div class="agent-states--grid" data-category="Static">
    <div class="agent-states--item">
      <p class="agent-states--label">:enabled</p>
      <scale-tag data-sketch-key="tag-22" variant="secondary"
        >Tag Label</scale-tag
      >
    </div>
    <div class="agent-states--item">
      <p class="agent-states--label">:hover</p>
      –
    </div>
    <div class="agent-states--item">
      <p class="agent-states--label">:focus</p>
      –
    </div>
    <div class="agent-states--item">
      <p class="agent-states--label">:active/:selected</p>
      –
    </div>
    <div class="agent-states--item">
      <p class="agent-states--label">:disabled</p>
      –
    </div>
  </div>

  <h2 class="title--artboard-section">Not Interactive – Colored – Small</h2>
  <div class="agent-states--grid" data-category="Static">
    <div class="agent-states--item">
      <p class="agent-states--label">:enabled</p>
      <scale-tag data-sketch-key="tag-23" variant="secondary" size="small"
        >Tag Label</scale-tag
      >
    </div>
    <div class="agent-states--item">
      <p class="agent-states--label">:hover</p>
      –
    </div>
    <div class="agent-states--item">
      <p class="agent-states--label">:focus</p>
      –
    </div>
    <div class="agent-states--item">
      <p class="agent-states--label">:active/:selected</p>
      –
    </div>
    <div class="agent-states--item">
      <p class="agent-states--label">:disabled</p>
      –
    </div>
  </div>
</div>

<script src="/symbol_names.js"></script>
<script>
  Array.from(document.querySelectorAll('scale-tag')).forEach((btn, i) => {
    btn.dataset.sketchSymbol = `Tag / ${order(
      1,
      getCategoryName(btn)
    )} / ${order(2, getSizeName(btn))} / ${order(
      3,
      btn.getAttribute('variant') ? 'Colored' : 'Black'
    )}${
      getCategoryName(btn) === 'Static'
        ? ''
        : ' / ' + order(4, getStateName(btn))
    }`;
  });
</script>
