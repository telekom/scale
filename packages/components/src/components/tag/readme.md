# t-tag



<!-- Auto Generated Below -->


## Properties

| Property      | Attribute      | Description                      | Type      | Default |
| ------------- | -------------- | -------------------------------- | --------- | ------- |
| `customClass` | `custom-class` | (optional) Tag class             | `string`  | `''`    |
| `link`        | `link`         | (optional) Tag on an <a> element | `string`  | `''`    |
| `pill`        | `pill`         | (optional) Tag pill              | `boolean` | `false` |
| `theme`       | `theme`        | (optional) Tag theme             | `string`  | `''`    |
| `variant`     | `variant`      | (optional) Tag variant           | `string`  | `''`    |


## CSS Custom Properties

| Name                                 | Description                                        |
| ------------------------------------ | -------------------------------------------------- |
| `--tag-background`                   | Background of the tag                              |
| `--tag-border`                       | Border of the tag                                  |
| `--tag-border-color`                 | Border color of the tag                            |
| `--tag-border-radius`                | Border radius of the tag                           |
| `--tag-color`                        | Color of the tag                                   |
| `--tag-font-size`                    | Font size of the tag                               |
| `--tag-link-active-background`       | Background color of the tag with link when active  |
| `--tag-link-focus-box-shadow`        | Box shadow of the tag with link when focus         |
| `--tag-link-focus-outline`           | outline of the tag with link when focus            |
| `--tag-link-hover-background`        | Background color of the tag with link when hover   |
| `--tag-link-text-decoration`         | Text decoration of the tag with link               |
| `--tag-link-visited-background`      | Background color of the tag with link when visited |
| `--tag-padding`                      | Padding of the tag                                 |
| `--tag-pill-border-radius`           | Large border radius of the tag pill                |
| `--tag-pill-padding`                 | Additional horizontal padding of the tag pill      |
| `--tag-variant-primary-background`   | Background of the tag variant primary              |
| `--tag-variant-primary-border-color` | Border color of the tag variant primary            |
| `--tag-variant-primary-color`        | Color of the tag variant primary                   |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
