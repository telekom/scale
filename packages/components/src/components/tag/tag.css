/**
* @prop --tag-color: Color of the tag
* @prop --tag-background: Background of the tag
* @prop --tag-border: Border of the tag
* @prop --tag-border-color: Border color of the tag
* @prop --tag-border-radius: Border radius of the tag
* @prop --tag-padding: Padding of the tag
* @prop --tag-font-size: Font size of the tag
* @prop --tag-variant-primary-color: Color of the tag variant primary
* @prop --tag-variant-primary-background: Background of the tag variant primary
* @prop --tag-variant-primary-border-color: Border color of the tag variant primary
* @prop --tag-pill-border-radius: Large border radius of the tag pill
* @prop --tag-pill-padding: Additional horizontal padding of the tag pill
* @prop --tag-link-text-decoration: Text decoration of the tag with link
* @prop --tag-link-hover-background: Background color of the tag with link when hover 
* @prop --tag-link-active-background: Background color of the tag with link when active 
* @prop --tag-link-visited-background: Background color of the tag with link when visited 
* @prop --tag-link-focus-outline: outline of the tag with link when focus 
* @prop --tag-link-focus-box-shadow: Box shadow of the tag with link when focus 
*/
.tag {
  display: inline-flex;
  color: var(--tag-color, #fff);
  background: var(--tag-background, #333);
  border: var(--tag-border, 1px solid);
  border-color: var(--tag-border-color, #dfdfdf);
  border-radius: var(--tag-border-radius, 0.25rem);
  padding: var(--tag-padding, 0.5rem);
  font-size: var(--tag-font-size, 0.8rem);
}

.tag--variant-primary {
  color: var(--tag-variant-primary-color, #fff);
  background: var(--tag-variant-primary-background, #007bff);
  border-color: var(--tag-variant-primary-border-color, #007bff);
}

.tag--pill {
  border-radius: var(--tag-pill-border-radius, 1rem);
  padding: var(--tag-pill-padding, 0.6rem);
}

.tag--link {
  text-decoration: var(--tag-link-text-decoration, none);
}

.tag--link:hover {
  background: var(--tag-link-hover-background, #1d2124);
}

.tag--link:active {
  background: var(--tag-link-active-background, #1d2124);
}

.tag--link:focus {
  outline: var(--tag-link-focus-outline, 0);
  box-shadow: var(
    --tag-link-focus-box-shadow,
    0 0 0 0.2rem rgba(52, 58, 64, 0.5)
  );
}
