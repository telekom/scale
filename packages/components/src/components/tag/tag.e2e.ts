import { newE2EPage } from '@stencil/core/testing';

describe('scale-tag', () => {
  it('renders', async () => {
    const page = await newE2EPage();
    await page.setContent('<scale-tag>Click me!</scale-tag>');
    const element = await page.find('scale-tag');
    expect(element).toHaveClass('hydrated');
  });
});
