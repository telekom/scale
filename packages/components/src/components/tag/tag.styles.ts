import { JssStyle } from 'jss';

const tagVariant = (variant: string) => ({
  [`tag--variant-${variant}`]: {
    color: ({ colors }) => colors[variant].contrastText,
    background: ({ colors }) => colors[variant].default,
    border: ({ colors }) => `1px solid ${colors[variant].default}`,
  },
});

const tagVariantLink = (variant: string) => ({
  [`&$tag--variant-${variant}`]: {
    ...tagVariant(variant),
    '&:hover': {
      color: ({ colors }) => colors[variant].contrastText,
      background: ({ colors }) => colors[variant].darker,
      border: ({ colors }) => `1px solid ${colors[variant].darker}`,
    },
    '&:active': {
      color: ({ colors }) => colors[variant].contrastText,
      background: ({ colors }) => colors[variant].darker,
      border: ({ colors }) => `1px solid ${colors[variant].darker}`,
    },
  },
});

export const styles: JssStyle = {
  tag: {
    display: 'inline-flex',
    alignItems: 'center',
    justifyContent: 'center',
    padding: '6px 12px',
    fontSize: 16,
    fontWeight: 400,
    lineHeight: 1,
    textAlign: 'center',
    whiteSpace: 'nowrap',
    verticalAlign: 'baseline',
    borderRadius: 4,
    transition: 'all 0.15s ease-in-out',
    background: '#343a40',
    color: ' #fff',
    '--icon-color': '#fff',
    '& scale-icon': {
      marginLeft: 2,
    },
    '&:not($tag--disabled)': {
      '& scale-icon': {
        '&:hover': {
          '--icon-color': 'red',
        },
      },
    },
  },
  'tag--dismissable': {
    paddingRight: 8,
  },
  'tag--size-small': {
    padding: '4px 8px',
    fontSize: 12,
    '& scale-icon': {
      height: 12,
      width: 12,
    },
    '&$tag--dismissable': {
      paddingRight: 4,
    },
  },
  ...tagVariant('primary'),
  ...tagVariant('secondary'),
  ...tagVariant('error'),
  ...tagVariant('warning'),
  ...tagVariant('info'),
  ...tagVariant('success'),
  'tag--link': {
    textDecoration: 'none',
    ...tagVariantLink('primary'),
    ...tagVariantLink('secondary'),
    ...tagVariantLink('error'),
    ...tagVariantLink('warning'),
    ...tagVariantLink('info'),
    ...tagVariantLink('success'),
  },
  'tag--disabled': {
    opacity: 0.5,
    cursor: 'not-allowed',
  },
};
