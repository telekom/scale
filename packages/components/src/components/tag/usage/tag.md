### Default

<scale-tag>
  Label
</scale-tag>

### Pill

<scale-tag pill>
  Label
</scale-tag>

### Link

<scale-tag href="http://example.com">
  Label
</scale-tag>
