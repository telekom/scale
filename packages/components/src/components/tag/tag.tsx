import { Component, Prop, h, Host, Event, EventEmitter } from '@stencil/core';
import { CssClassMap } from '../../utils/utils';
import classNames from 'classnames';
import { styles } from './tag.styles';
import { CssInJs } from '../../utils/css-in-js';
import { StyleSheet } from 'jss';
import Base from '../../utils/base-interface';
import { defaultTheme } from '../../theme/defaultTheme';

@Component({
  tag: 'scale-tag',
  shadow: true,
})
export class Tag implements Base {
  /** (optional) Tag class */
  @Prop() customClass?: string = '';
  /** (optional) Tag size */
  @Prop() size?: string = '';
  /** (optional) Tag variant */
  @Prop() variant?: string = '';
  /** (optional) Tag href */
  @Prop() href?: string = '';
  /** (optional) Tag target */
  @Prop() target?: string = '_self';
  /** (optional) Tag dismissable */
  @Prop() dismissable?: boolean = false;
  /** (optional) Tag disabled */
  @Prop() disabled?: boolean = false;

  /** (optional) Injected jss styles */
  @Prop() styles?: StyleSheet;
  /** decorator Jss stylesheet */
  @CssInJs('Tag', styles) stylesheet: StyleSheet;

  @Event() close: EventEmitter<any>;

  componentWillLoad() {}
  componentWillUpdate() {}

  handleClose(event) {
    event.preventDefault();
    event.stopPropagation();
    this.close.emit(event);
  }

  render() {
    const Element = !!this.href && !this.disabled ? 'a' : 'span';
    const linkProps = !!this.href
      ? {
          href: this.href,
          target: this.target,
        }
      : {};
    const iconProps = !this.disabled
      ? {
          focusable: true,
          onClick: event => this.handleClose(event),
        }
      : {};
    const theme = (window as any).scale && (window as any).scale.theme;
    const { icons } = theme || defaultTheme;
    return (
      <Host>
        <style>{this.stylesheet.toString()}</style>
        <Element class={this.getCssClassMap()} {...linkProps}>
          <slot />
          {this.dismissable && (
            <scale-icon size={16} path={icons.close} {...iconProps} />
          )}
        </Element>
      </Host>
    );
  }

  getCssClassMap(): CssClassMap {
    const { classes } = this.stylesheet;
    return classNames(
      classes.tag,
      this.customClass && this.customClass,
      this.size && classes[`tag--size-${this.size}`],
      this.variant && classes[`tag--variant-${this.variant}`],
      !!this.href && classes[`tag--link`],
      !!this.dismissable && classes[`tag--dismissable`],
      !!this.disabled && classes[`tag--disabled`]
    );
  }
}
