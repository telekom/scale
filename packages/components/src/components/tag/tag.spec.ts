import { newSpecPage } from '@stencil/core/testing';
import { Tag } from './tag';
import { styles } from './tag.styles';
import jss from 'jss';

describe('Tag', () => {
  let element;
  let stylesheet;
  beforeEach(async () => {
    element = new Tag();
    stylesheet = element.stylesheet = jss.createStyleSheet(styles as any);
  });

  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [Tag],
      html: `<scale-tag>Label</scale-tag>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should have a link', async () => {
    const page = await newSpecPage({
      components: [Tag],
      html: `<scale-tag href="#">Label</scale-tag>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should handle css classes', () => {
    element.customClass = 'custom';
    expect(element.getCssClassMap()).toContain('custom');

    element.size = 'small';
    stylesheet.addRule('tag--size-small', {});
    expect(element.getCssClassMap()).toContain(
      stylesheet.classes['tag--size-small']
    );

    element.variant = 'primary';
    stylesheet.addRule('tag--variant-primary', {});
    expect(element.getCssClassMap()).toContain(
      stylesheet.classes['tag--variant-primary']
    );

    element.href = 'http://example.com';
    expect(element.getCssClassMap()).toContain(stylesheet.classes['tag--link']);
  });
});
