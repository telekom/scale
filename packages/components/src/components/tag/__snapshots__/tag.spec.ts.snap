// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag should have a link 1`] = `
<scale-tag href="#">
  <mock:shadow-root>
    <style>
      .tag-0-0-34 {
  color:  #fff;
  display: inline-flex;
  padding: 6px 12px;
  font-size: 16px;
  background: #343a40;
  text-align: center;
  transition: all 0.15s ease-in-out;
  align-items: center;
  font-weight: 400;
  line-height: 1;
  white-space: nowrap;
  --icon-color: #fff;
  border-radius: 4px;
  vertical-align: baseline;
  justify-content: center;
}
.tag-0-0-34 scale-icon {
  margin-left: 2px;
}
.tag-0-0-34:not(.tag--disabled-0-0-44) {}
.tag-0-0-34:not(.tag--disabled-0-0-44) scale-icon {}
.tag-0-0-34:not(.tag--disabled-0-0-44) scale-icon:hover {
  --icon-color: red;
}
.tag--dismissable-0-0-35 {
  padding-right: 8px;
}
.tag--size-small-0-0-36 {
  padding: 4px 8px;
  font-size: 12px;
}
.tag--size-small-0-0-36 scale-icon {
  width: 12px;
  height: 12px;
}
.tag--size-small-0-0-36.tag--dismissable-0-0-35 {
  padding-right: 4px;
}
.tag--variant-primary-0-0-37 {
  color: #fff;
  background: #2962ff;
  border: 1px solid #2962ff;
}
.tag--variant-secondary-0-0-38 {
  color: #fff;
  background: #d500f9;
  border: 1px solid #d500f9;
}
.tag--variant-error-0-0-39 {
  color: #fff;
  background: #f50057;
  border: 1px solid #f50057;
}
.tag--variant-warning-0-0-40 {
  color: #fff;
  background: #ffc400;
  border: 1px solid #ffc400;
}
.tag--variant-info-0-0-41 {
  color: #fff;
  background: #00b0ff;
  border: 1px solid #00b0ff;
}
.tag--variant-success-0-0-42 {
  color: #fff;
  background: #00e676;
  border: 1px solid #00e676;
}
.tag--link-0-0-43 {
  text-decoration: none;
}
.tag--link-0-0-43.tag--variant-primary-0-0-37 {}
.tag--link-0-0-43.tag--variant-secondary-0-0-38 {}
.tag--link-0-0-43.tag--variant-error-0-0-39 {}
.tag--link-0-0-43.tag--variant-warning-0-0-40 {}
.tag--link-0-0-43.tag--variant-info-0-0-41 {}
.tag--link-0-0-43.tag--variant-success-0-0-42 {}
.tag--link-0-0-43.tag--variant-success-0-0-42:hover {
  color: #fff;
  background: #00b248;
  border: 1px solid #00b248;
}
.tag--link-0-0-43.tag--variant-success-0-0-42:active {
  color: #fff;
  background: #00b248;
  border: 1px solid #00b248;
}
.tag--link-0-0-43.tag--variant-info-0-0-41:hover {
  color: #fff;
  background: #0081cb;
  border: 1px solid #0081cb;
}
.tag--link-0-0-43.tag--variant-info-0-0-41:active {
  color: #fff;
  background: #0081cb;
  border: 1px solid #0081cb;
}
.tag--link-0-0-43.tag--variant-warning-0-0-40:hover {
  color: #fff;
  background: #c79400;
  border: 1px solid #c79400;
}
.tag--link-0-0-43.tag--variant-warning-0-0-40:active {
  color: #fff;
  background: #c79400;
  border: 1px solid #c79400;
}
.tag--link-0-0-43.tag--variant-error-0-0-39:hover {
  color: #fff;
  background: #bb002f;
  border: 1px solid #bb002f;
}
.tag--link-0-0-43.tag--variant-error-0-0-39:active {
  color: #fff;
  background: #bb002f;
  border: 1px solid #bb002f;
}
.tag--link-0-0-43.tag--variant-secondary-0-0-38:hover {
  color: #fff;
  background: #9e00c5;
  border: 1px solid #9e00c5;
}
.tag--link-0-0-43.tag--variant-secondary-0-0-38:active {
  color: #fff;
  background: #9e00c5;
  border: 1px solid #9e00c5;
}
.tag--link-0-0-43.tag--variant-primary-0-0-37:hover {
  color: #fff;
  background: #0039cb;
  border: 1px solid #0039cb;
}
.tag--link-0-0-43.tag--variant-primary-0-0-37:active {
  color: #fff;
  background: #0039cb;
  border: 1px solid #0039cb;
}
.tag--disabled-0-0-44 {
  cursor: not-allowed;
  opacity: 0.5;
}
    </style>
    <a class="tag--link-0-0-43 tag-0-0-34" href="#" target="_self">
      <slot></slot>
    </a>
  </mock:shadow-root>
  Label
</scale-tag>
`;

exports[`Tag should match snapshot 1`] = `
<scale-tag>
  <mock:shadow-root>
    <style>
      .tag-0-0-12 {
  color:  #fff;
  display: inline-flex;
  padding: 6px 12px;
  font-size: 16px;
  background: #343a40;
  text-align: center;
  transition: all 0.15s ease-in-out;
  align-items: center;
  font-weight: 400;
  line-height: 1;
  white-space: nowrap;
  --icon-color: #fff;
  border-radius: 4px;
  vertical-align: baseline;
  justify-content: center;
}
.tag-0-0-12 scale-icon {
  margin-left: 2px;
}
.tag-0-0-12:not(.tag--disabled-0-0-22) {}
.tag-0-0-12:not(.tag--disabled-0-0-22) scale-icon {}
.tag-0-0-12:not(.tag--disabled-0-0-22) scale-icon:hover {
  --icon-color: red;
}
.tag--dismissable-0-0-13 {
  padding-right: 8px;
}
.tag--size-small-0-0-14 {
  padding: 4px 8px;
  font-size: 12px;
}
.tag--size-small-0-0-14 scale-icon {
  width: 12px;
  height: 12px;
}
.tag--size-small-0-0-14.tag--dismissable-0-0-13 {
  padding-right: 4px;
}
.tag--variant-primary-0-0-15 {
  color: #fff;
  background: #2962ff;
  border: 1px solid #2962ff;
}
.tag--variant-secondary-0-0-16 {
  color: #fff;
  background: #d500f9;
  border: 1px solid #d500f9;
}
.tag--variant-error-0-0-17 {
  color: #fff;
  background: #f50057;
  border: 1px solid #f50057;
}
.tag--variant-warning-0-0-18 {
  color: #fff;
  background: #ffc400;
  border: 1px solid #ffc400;
}
.tag--variant-info-0-0-19 {
  color: #fff;
  background: #00b0ff;
  border: 1px solid #00b0ff;
}
.tag--variant-success-0-0-20 {
  color: #fff;
  background: #00e676;
  border: 1px solid #00e676;
}
.tag--link-0-0-21 {
  text-decoration: none;
}
.tag--link-0-0-21.tag--variant-primary-0-0-15 {}
.tag--link-0-0-21.tag--variant-secondary-0-0-16 {}
.tag--link-0-0-21.tag--variant-error-0-0-17 {}
.tag--link-0-0-21.tag--variant-warning-0-0-18 {}
.tag--link-0-0-21.tag--variant-info-0-0-19 {}
.tag--link-0-0-21.tag--variant-success-0-0-20 {}
.tag--link-0-0-21.tag--variant-success-0-0-20:hover {
  color: #fff;
  background: #00b248;
  border: 1px solid #00b248;
}
.tag--link-0-0-21.tag--variant-success-0-0-20:active {
  color: #fff;
  background: #00b248;
  border: 1px solid #00b248;
}
.tag--link-0-0-21.tag--variant-info-0-0-19:hover {
  color: #fff;
  background: #0081cb;
  border: 1px solid #0081cb;
}
.tag--link-0-0-21.tag--variant-info-0-0-19:active {
  color: #fff;
  background: #0081cb;
  border: 1px solid #0081cb;
}
.tag--link-0-0-21.tag--variant-warning-0-0-18:hover {
  color: #fff;
  background: #c79400;
  border: 1px solid #c79400;
}
.tag--link-0-0-21.tag--variant-warning-0-0-18:active {
  color: #fff;
  background: #c79400;
  border: 1px solid #c79400;
}
.tag--link-0-0-21.tag--variant-error-0-0-17:hover {
  color: #fff;
  background: #bb002f;
  border: 1px solid #bb002f;
}
.tag--link-0-0-21.tag--variant-error-0-0-17:active {
  color: #fff;
  background: #bb002f;
  border: 1px solid #bb002f;
}
.tag--link-0-0-21.tag--variant-secondary-0-0-16:hover {
  color: #fff;
  background: #9e00c5;
  border: 1px solid #9e00c5;
}
.tag--link-0-0-21.tag--variant-secondary-0-0-16:active {
  color: #fff;
  background: #9e00c5;
  border: 1px solid #9e00c5;
}
.tag--link-0-0-21.tag--variant-primary-0-0-15:hover {
  color: #fff;
  background: #0039cb;
  border: 1px solid #0039cb;
}
.tag--link-0-0-21.tag--variant-primary-0-0-15:active {
  color: #fff;
  background: #0039cb;
  border: 1px solid #0039cb;
}
.tag--disabled-0-0-22 {
  cursor: not-allowed;
  opacity: 0.5;
}
    </style>
    <span class="tag-0-0-12">
      <slot></slot>
    </span>
  </mock:shadow-root>
  Label
</scale-tag>
`;
